backend/de.metas.business/src/main/java/de/metas/pricing/service/impl/PriceListDAO.java

```diff
@@ -121,8 +121,23 @@ public I_M_PricingSystem getPricingSystemById(final PricingSystemId pricingSyste
 		return loadOutOfTrx(pricingSystemId, I_M_PricingSystem.class);
 	}
 
+	@NonNull
 	@Override
 	public PricingSystemId getPricingSystemIdByValue(@NonNull final String value)
+	{
+		final PricingSystemId pricingSystemId = getPricingSystemIdByValueOrNull(value);
+
+		if (pricingSystemId == null)
+		{
+			throw new AdempiereException("@NotFound@ @M_PricingSystem_ID@ (@Value@=" + value + ")");
+		}
+
+		return pricingSystemId;
+	}
+
+	@Nullable
+	@Override
+	public PricingSystemId getPricingSystemIdByValueOrNull(@NonNull final String value)
 	{
 		final int pricingSystemId = Services.get(IQueryBL.class)
 				.createQueryBuilderOutOfTrx(I_M_PricingSystem.class)
```

Comment:
since it's not unique by AD_Client_ID i think you need to ider by AD_Client_ID desc and get the first match. also, imo filter for active-records-only

Reply:
makes sense.